app/src/main/java/com/example/trainingintive/viewmodels/SplashViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.trainingintive.viewmodels
+
+import androidx.lifecycle.ViewModel
+import com.example.trainingintive.navigators.Navigator
+import com.example.trainingintive.navigators.SplashNavigator
+import com.example.trainingintive.util.Event
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.disposables.Disposable
+import io.reactivex.rxjava3.schedulers.Schedulers
+import java.util.concurrent.TimeUnit
+
+class SplashViewModel : ViewModel() {
+
+    val navigator: Navigator = SplashNavigator()
+    val downloadData: Disposable
+
+    init {
+        downloadData = Single.timer(5, TimeUnit.SECONDS)
+            .doOnSuccess({ navigator.setEvent(Event.Downloading) })
+            .flatMap {
+                Single.fromCallable({ downloadData() })
```

Comment:
You can dropped it, it doesn't make sense to keep it since you use `timer` to simulate delay